test/com/ableton/JenkinsMocks.groovy

```diff
@@ -17,4 +17,74 @@ class JenkinsMocks {
       return System.properties['user.dir']
     }
   }
+
+  /**
+   * Simple container for holding mock script output.
+   * @see JenkinsMocks#addShMock
+   */
+  class MockScriptOutput {
+    String stdout
+    int exitValue
+
+    MockScriptOutput(String stdout, int exitValue) {
+      this.stdout = stdout
+      this.exitValue = exitValue
+    }
+  }
+
+  /** Holds configured mock output values for the `sh` command. */
+  static Map<String, MockScriptOutput> mockScripts = [:]
+
+  /**
+   * Configure mock output for the `sh` command. This function should be called before
+   * attempting to call JenkinsMocks.sh.
+   * @see JenkinsMocks#sh
+   * @param script Script command to mock.
+   * @param stdout Standard output text to return for the given command.
+   * @param exitValue Exit value for the command (default: 0).
+   * @return
+   */
+  static addShMock(String script, String stdout, int exitValue = 0) {
+    mockScripts[script] = new MockScriptOutput(null, stdout, exitValue)
+  }
+
+  static Closure sh = { args ->
+    String script = null
+    def returnStdout = false
+    def returnStatus = false
+
+    // The `sh` function can be called with either a string, or a map of key/value pairs.
+    if (args instanceof String || args instanceof GString) {
```

Comment:
Out of curiosity, what the hell are `GString`s?

Reply:
When you define a string with `'hello'`, it's a `java.lang.String`, but if you want to use interpolation (ie, `"hello ${thing}"`), then it's a Groovy GString. See:

http://mrhaki.blogspot.se/2009/08/groovy-goodness-string-strings-strings.html

Reply:
Thanks!